src/components_new/DocumentEditor.tsx

```
@@ -182,6 +197,48 @@ export function DocumentEditor({ document: doc, initialContent, isReadOnly }: Ed
+    setReplacementSuggestion(null);
+  };
+
+  const handleTextReplace = () => {
+    if (!editor) return;
+    const pos = editor.state.selection.from;
+    const textBefore = editor.state.doc.textBetween(0, pos, ' ');
+    const match = /\S+$/.exec(textBefore);
+    if (!match) return;
+    const lastWord = match[0];
+    const entry = dictionary.find(
+      (e: any) => e.from === lastWord || (lastWord.startsWith('@') && e.from === lastWord.slice(1))
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)